nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -242,6 +243,9 @@ Costmap2DROS::on_activate(const rclcpp_lifecycle::State & /*state*/)
   stopped_ = true;  // to active plugins
   stop_updates_ = false;
   map_update_thread_shutdown_ = false;
+
+  updateMapOnRequest();
+  map_update_frequency_ = update_on_request_ ? 0.0 : map_update_frequency_;
```

Comment:
That makes sense, I must have missed it. Thanks for highlighting this

Reply:
Also, are you able to clarify what you mean by moving the thread creation/joining into a separate sub-routine? Any example would be helpful

Reply:
I mean some simple routine, with code like:
```
Costmap2DROS::mapUpdateThreadOn() {
  map_update_thread_shutdown_ = false;
  map_update_thread_ = std::make_unique<std::thread>(
    std::bind(&Costmap2DROS::mapUpdateLoop, this, map_update_frequency_));
}

Costmap2DROS::mapUpdateThreadOff() {
  map_update_thread_shutdown_ = true;
  map_update_thread_->join();

}
```
Or whenever you want.
Since these are only two lines, it is not mandatory: you could add these lines just directly in all required places (`on_activate()`, `on_deactivate()` and `dynamicParametersCallback()`).

Reply:
Okay I see what you mean. Yeah I think a routine would make things much clearer, I will incorporate this